From 754e0c6112a181b30bd46aa13812f37db01c3387 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 10 May 2016 15:09:26 +0800
Subject: [PATCH] pcm: Don't store the state for SND_PCM_STATE_SUSPENDED

The resume function don't update the dmix->state, if store SUSPENDED
state in snd_pcm_dmix_state, the write function after resume will
return error -ESTRPIPE, because the snd_pcm_write_areas() will check
the state of the pcm device.
This patch remove the store SND_PCM_STATE_SUSPENDED state operation
for dmix,dshare,dsnoop.

Upstream-Status: Pending

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>

---
 src/pcm/pcm_dmix.c   | 2 +-
 src/pcm/pcm_dshare.c | 2 +-
 src/pcm/pcm_dsnoop.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/pcm/pcm_dmix.c b/src/pcm/pcm_dmix.c
index 5b7472d..e08e46f 100644
--- a/src/pcm/pcm_dmix.c
+++ b/src/pcm/pcm_dmix.c
@@ -460,9 +460,9 @@ static snd_pcm_state_t snd_pcm_dmix_state(snd_pcm_t *pcm)
 	snd_pcm_state_t state;
 	state = snd_pcm_state(dmix->spcm);
 	switch (state) {
-	case SND_PCM_STATE_SUSPENDED:
 	case SND_PCM_STATE_DISCONNECTED:
 		dmix->state = state;
+	case SND_PCM_STATE_SUSPENDED:
 		return state;
 	case SND_PCM_STATE_XRUN:
 		if ((err = snd_pcm_direct_slave_recover(dmix)) < 0)
diff --git a/src/pcm/pcm_dshare.c b/src/pcm/pcm_dshare.c
index 8a67257..8ba61b4 100644
--- a/src/pcm/pcm_dshare.c
+++ b/src/pcm/pcm_dshare.c
@@ -258,9 +258,9 @@ static snd_pcm_state_t snd_pcm_dshare_state(snd_pcm_t *pcm)
 	snd_pcm_state_t state;
 	state = snd_pcm_state(dshare->spcm);
 	switch (state) {
-	case SND_PCM_STATE_SUSPENDED:
 	case SND_PCM_STATE_DISCONNECTED:
 		dshare->state = state;
+	case SND_PCM_STATE_SUSPENDED:
 		return state;
 	case SND_PCM_STATE_XRUN:
 		if ((err = snd_pcm_direct_slave_recover(dshare)) < 0)
diff --git a/src/pcm/pcm_dsnoop.c b/src/pcm/pcm_dsnoop.c
index c6e8cd2..9610482 100644
--- a/src/pcm/pcm_dsnoop.c
+++ b/src/pcm/pcm_dsnoop.c
@@ -208,9 +208,9 @@ static snd_pcm_state_t snd_pcm_dsnoop_state(snd_pcm_t *pcm)
 	snd_pcm_state_t state;
 	state = snd_pcm_state(dsnoop->spcm);
 	switch (state) {
-	case SND_PCM_STATE_SUSPENDED:
 	case SND_PCM_STATE_DISCONNECTED:
 		dsnoop->state = state;
+	case SND_PCM_STATE_SUSPENDED:
 		return state;
 	case SND_PCM_STATE_XRUN:
 		if ((err = snd_pcm_direct_slave_recover(dsnoop)) < 0)
